pasarguard.sh

```
@@ -2504,39 +2504,87 @@ restore_command() {
+                # TimescaleDB version because DROP EXTENSION / CREATE EXTENSION cycles
+                # break when the shared library is already loaded with the new version.
+                # The fix: drop & recreate the database, then use the official
+                # timescaledb_pre_restore() / timescaledb_post_restore() wrapper.
+                # See: https://docs.timescale.com/self-hosted/latest/backup-and-restore/
+                colorized_echo blue "Using TimescaleDB-safe restore procedure..."
+
+                # Drop and recreate the target database for a clean slate
+                colorized_echo blue "Dropping and recreating database '$db_name'..."
+                docker exec "$container_name" psql -U postgres -d postgres \
+                    -c "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname = '$db_name' AND pid <> pg_backend_pid();" \
+                    >>"$log_file" 2>&1
```

Shellcheck (0.11.0)

[style] Consider using { cmd1; cmd2; } >> file instead of individual redirects.

(SC2129)